base/compiler/ssair/verify.jl

```diff
@@ -1,9 +1,29 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
 function maybe_show_ir(ir::IRCode)
-    if isdefined(Core, :Main)
+    if isdefined(Core, :Main) && isdefined(Core.Main, :Base)
         # ensure we use I/O that does not yield, as this gets called during compilation
         invokelatest(Core.Main.Base.show, Core.stdout, "text/plain", ir)
+    else
+        print(Core.stderr, "Verification error in IR for function ")
+        print(Core.stderr, ir.argtypes[1])
+        print(Core.stderr, " with args ")
+        println(Core.stderr, ir.argtypes[2:end])
```

Comment:
This should probably either do its own array printing like you do below, or we should go ahead and finally fix how `Array` is printed in `Core`